src/memo/memo.ml

```diff
@@ -517,10 +583,20 @@ let create (type i o f) name ~doc ~input:(module Input : Input with type t = i)
   ( match visibility with
   | Public _ -> Spec.register spec
   | Hidden -> () );
-  let cache = Table.create (module Input) 1024 in
-  Caches.register ~clear:(fun () -> Table.clear cache);
+  Caches.register ~clear:(fun () -> Store.clear cache);
   { cache; spec }
 
+let create_with_store (type i) name ~store:(module S : Store with type key = i)
+    ~doc ~input ~visibility ~output typ f =
+  let cache = Store.make (module S) in
+  create_with_cache name ~cache ~doc ~input ~output ~visibility typ f
+
+let create (type i) name ~doc ~input:(module Input : Input with type t = i)
+    ~visibility ~output typ f =
+  let cache = Store.of_table (Table.create (module Input) 1024) in
```

Comment:
I never noticed this before, but 1024 is a bit ridiculous default size. It shouldn't be above 10 or so.

Reply:
I pushed that change too.